CollAction/Controllers/ManageController.cs

```diff
@@ -176,6 +190,20 @@ public async Task<IActionResult> SetPassword(SetPasswordViewModel model)
             return RedirectToAction(nameof(Index), new { Message = ManageMessageId.Error });
         }
 
+        [HttpPost]
+        [Authorize]
```

Comment:
Good idea to validate the anti-forgery token as well. You can use it within javascript: https://docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?view=aspnetcore-2.2#javascript-ajax-and-spas

Reply:
Done.